pkg/controller/backup_controller.go

```diff
@@ -535,6 +544,32 @@ func (c *backupController) runBackup(backup *pkgbackup.Request) error {
 		fatalErrs = append(fatalErrs, err)
 	}
 
+	// Empty slices here so that they can be passed in to the persistBackup call later, regardless of whether or not CSI's enabled.
+	// This way, we only make the Lister call if the feature flag's on.
+	var volumeSnapshots []*snapshotv1beta1api.VolumeSnapshot
+	var volumeSnapshotContents []*snapshotv1beta1api.VolumeSnapshotContent
+	if features.IsEnabled(velerov1api.CSIFeatureFlag) {
+		selector := labels.SelectorFromSet(map[string]string{velerov1api.BackupNameLabel: backup.Name})
+
+		// Listers are wrapped in a nil check out of caution, since they may not be populated based on the
+		// EnableCSI feature flag. This is more to guard against programmer error, as they shouldn't be nil
+		// when EnableCSI is on.
+		if c.volumeSnapshotLister != nil {
+			volumeSnapshots, err = c.volumeSnapshotLister.List(selector)
+			if err != nil {
+				backupLog.Error(err)
+			}
+		}
+
+		if c.volumeSnapshotContentLister != nil {
+			volumeSnapshotContents, err = c.volumeSnapshotContentLister.List(selector)
```

Comment:
one alternative to requiring a label selector for the VolumeSnapshotContents would be to iterate through `volumeSnapshots`, find which VSC each one is bound to, and then explicitly grab those VSCs from the Lister. Not sure if it makes the flow easier or not - just an option to consider.

Reply:
I'll see what this looks like. I'm not 100% sure if we even need to upload the VSCs, honestly, but since the rest of the code is there to do it, might as well give it a shot.

Also, I think regardless of implementation here, it would be useful to have the VSCs labeled with the Velero backup name, to indicated Velero made them.

Reply:
Yep, that worked 👍